pkg/appgw/backendhttpsettings.go

```diff
@@ -133,7 +135,7 @@ func (c *appGwConfigBuilder) BackendHTTPSettingsCollection(ingressList []*v1beta
 	}
 
 	if len(unresolvedBackendID) > 0 {
-		return errors.New("unable to resolve backend port for some services")
+		return nil, backendHTTPSettingsMap, finalServiceBackendPairMap, errors.New("unable to resolve backend port for some services")
```

Comment:
Since we have reached an error state here - I'd return `nil` for the first three items (not that it really matters)